frontend/src/hooks/useGaUserData.js

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'react';
+
+export default function useGaUserData(user) {
+  useEffect(() => {
+    try {
+      if (!user) {
+        return;
+      }
+      if (window.dataLayer && Array.isArray(window.dataLayer)) {
+        const eventNames = window.dataLayer.map((item) => item.event);
+        if (eventNames.includes('userData')) {
+          return;
+        }
+
+        if (!user.id || !user.roles) {
```

Comment:
Is there reason we dont have this also on line 6.

if (!user && (!user.id || !user.roles))

Reply:
I condensed the if statements and added some comments https://github.com/HHS/Head-Start-TTADP/pull/1755/commits/c11b8c0377225eb8bf09fb1c7476920e816b921f